src/psyclone/psyir/frontend/fortran.py

```diff
@@ -83,6 +83,11 @@ def __init__(self, free_form: bool = True, ignore_comments: bool = True,
         if not self._parser:
             self._parser = ParserFactory().create(std="f2008")
         self._free_form = free_form
+        if ignore_comments and not ignore_directives:
+            raise ValueError(
```

Comment:
Please document this with a new ":raises ValueError:" in the docstring.

Reply:
Ah yes, good point. Addressed in 347af2ec2.